File: javascript/binary-search-tree/binary-search-tree.js
var BinaryTree = function(data) {
	/* Tree data structure */
  this.data = data;
  this.left = null;
  this.right = null;
};

BinaryTree.prototype.insert = function(data) {
	/* Adds a value to the tree */
  var node = (data <= this.data) ? 'left' : 'right';
  if( this[node] )
		this[node].insert(data);
	else
		this[node] = new BinaryTree(data);

};

BinaryTree.prototype.each = function(func) {
	/* Sorted elements in the tree */

};

module.exports = BinaryTree;

File: ruby/binary-search-tree/binary_search_tree.rb
class Bst
  attr_reader :data, :left, :right
  def initialize(value)
    @data  = value
    @left  = nil
    @right = nil
  end

  def insert(value)
    if value <= @data
      @left.insert(value) if @left != nil
      @left = Bst.new(value) if @left == nil
    else
      @right.insert(value) if @right != nil
      @right = Bst.new(value) if @right == nil
    end
  end

end

module BookKeeping
  Bst::VERSION = 1
end
